### PR TITLE
Add retry field for builds

### PR DESCRIPTION
**Pull request by tetigi**

### CODE REVIEW

File: src/kraken/std/cargo/tasks/cargo_build_task.py [hunk 1]
```diff
@@ -93,12 +96,20 @@ def execute(self) -> TaskStatus:
...
+            if result == 0:
+                for out_bin in out_binaries:
+                    assert out_bin.path.is_file(), out_bin
+                for out_lib in out_libraries:
+                    assert out_lib.path.is_file(), out_lib
+                break
+            else:
+                total_attempts -= 1
+                self.logger.warn("%s failed with result %s", safe_command, result)
+                self.logger.warn("There are %s attempts remaining", total_attempts)
```
Comment (wngr) [inline src/kraken/std/cargo/tasks/cargo_build_task.py:113]: Maybe add some wait in between.

Comment (tetigi) [reply src/kraken/std/cargo/tasks/cargo_build_task.py:114]: Added 10 secs.

File: src/kraken/std/cargo/tasks/cargo_build_task.py [hunk 2]
```diff
@@ -93,12 +97,23 @@ def execute(self) -> TaskStatus:
...
+            result = sp.call(command, cwd=self.project.directory, env={**os.environ, **env})
+
+            if result == 0:
+                for out_bin in out_binaries:
+                    assert out_bin.path.is_file(), out_bin
+                for out_lib in out_libraries:
+                    assert out_lib.path.is_file(), out_lib
+                break
+            else:
+                total_attempts -= 1
```
Comment (NiklasRosenstein) [inline src/kraken/std/cargo/tasks/cargo_build_task.py:112]: Should we sleep for a bit at least?

Comment (NiklasRosenstein) [reply src/kraken/std/cargo/tasks/cargo_build_task.py:112]: Ah actually there's a sleep below.

**Pull request merged**